app/models/compute_resource.rb

```diff
@@ -448,9 +460,13 @@ def nested_attributes_for(type, opts)
     end.compact
   end
 
-  def associate_by(name, attributes)
+  def associate_by(name, attributes, host = nil)
```

Comment:
How about creating a new method `associate_by_host`?

Reply:
Hey @stejskalleos,
this is exactly what my is supposed to do. And thanks for your suggestion with the new `associate_by_host` method. I adjusted my changes accordlingly and found out that this approach makes it all in all really simple.
IMO no refactoring is needed anymore. What do you think?